### PR TITLE
fix(backend): Handle token type array in authenticateRequest

### PR DESCRIPTION
**Pull request by wobsoriano**
## Description

Cherry-picked changes from https://github.com/clerk/javascript/pull/7556

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for token authentication, including scenarios for session tokens, machine tokens, and mixed token types in various authentication methods (header, cookies).
  * Added tests for unauthenticated states and edge cases.

* **Bug Fixes**
  * Improved token handling in authentication requests to support array-type token configurations in header-based authentication.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .changeset/weak-wombats-begin.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
[the remark this answers is not in the document]
Comment (jacekradko) [reply .changeset/weak-wombats-begin.md:2]: @wobsoriano We should have a changeset for the backend pacakge

[the remark this answers is not in the document]
Comment (wobsoriano) [reply .changeset/weak-wombats-begin.md:2]: This was released in core 2 as well. Do we still need it here?

[the remark this answers is not in the document]
Comment (jacekradko) [reply .changeset/weak-wombats-begin.md:2]: That's fair. We haven't really split the changesets this way for many of the core-2 + core-3 features so it feels odd to have this empty. All good though
